// @koumoul/vuetify-jsonschema-form/doc/.nuxt/router.js
import Vue from 'vue'
import Router from 'vue-router'
import { interopDefault } from './utils'
import scrollBehavior from './router.scrollBehavior.js'

const _4e863f76 = () => interopDefault(import('../pages/examples/index.vue' /* webpackChunkName: "pages/examples/index" */))
const _1269a75e = () => interopDefault(import('../pages/examples/_id.vue' /* webpackChunkName: "pages/examples/_id" */))
const _74a27ad8 = () => interopDefault(import('../pages/index.vue' /* webpackChunkName: "pages/index" */))

// TODO: remove in Nuxt 3
const emptyFn = () => {}
const originalPush = Router.prototype.push
Router.prototype.push = function push (location, onComplete = emptyFn, onAbort) {
  return originalPush.call(this, location, onComplete, onAbort)
}

Vue.use(Router)

export const routerOptions = {
  mode: 'history',
  base: decodeURI('/'),
  linkActiveClass: 'nuxt-link-active',
  linkExactActiveClass: 'nuxt-link-exact-active',
  scrollBehavior,

  routes: [{
    path: "/examples",
    component: _4e863f76,
    name: "examples"
  }, {
    path: "/examples/:id",
    component: _1269a75e,
    name: "examples-id"
  }, {
    path: "/",
    component: _74a27ad8,
    name: "index"
  }],

  fallback: false
}

export function createRouter () {
  return new Router(routerOptions)
}
